src/main/java/org/mockito/internal/junit/JUnitRule.java

```diff
@@ -8,38 +8,49 @@
 import org.junit.runners.model.Statement;
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
+import org.mockito.exceptions.base.MockitoAssertionError;
 import org.mockito.internal.util.MockitoLogger;
+import org.mockito.internal.util.SimpleMockitoLogger;
 import org.mockito.junit.MockitoRule;
 
 /**
  * Internal implementation.
  */
 public class JUnitRule implements MockitoRule {
-	
+
+    public enum Strictness { SILENT, WARN, ERROR }
 	private final MockitoLogger logger;
-    private final boolean silent;
+    private final Strictness strictness;
 
     /**
      * @param logger target for the stubbing warnings
-     * @param silent whether the rule
+     * @param strictness whether the rule
      */
-    public JUnitRule(MockitoLogger logger, boolean silent) {
+    public JUnitRule(MockitoLogger logger, Strictness strictness) {
 		this.logger = logger;
-        this.silent = silent;
+        this.strictness = strictness;
     }
 
 	@Override
 	public Statement apply(final Statement base, FrameworkMethod method, final Object target) {
-        if (silent) {
+        if (strictness == Strictness.SILENT) {
```

Comment:
Lets do `strictness.applyStatement(...)` here. Makes it easier to add a new enum value for later.